controllers/kustomization_decryptor.go

```diff
@@ -136,18 +138,25 @@ func (kd *KustomizeDecryptor) ImportKeys(ctx context.Context) error {
 		}
 		defer os.RemoveAll(tmpDir)
 
-		for name, key := range secret.Data {
-			keyPath, err := securejoin.SecureJoin(tmpDir, name)
-			if err != nil {
-				return err
-			}
-			if err := ioutil.WriteFile(keyPath, key, os.ModePerm); err != nil {
-				return fmt.Errorf("unable to write key to storage: %w", err)
-			}
-			if err := kd.gpgImport(keyPath); err != nil {
-				return err
+		var ageIdentities []string
+		for name, file := range secret.Data {
+			switch filepath.Ext(name) {
+			case ".asc":
+				keyPath, err := securejoin.SecureJoin(tmpDir, name)
+				if err != nil {
+					return err
+				}
+				if err := ioutil.WriteFile(keyPath, file, os.ModePerm); err != nil {
+					return fmt.Errorf("unable to write key to storage: %w", err)
+				}
+				if err := kd.gpgImport(keyPath); err != nil {
+					return err
+				}
+			case ".txt":
```

Comment:
Wondering if this very generic extension is going to bite us later if some other encryption format is introduced, but do not have a clue about good alternatives, given `.age` is reserved for encrypted age files (and not keys).

Reply:
What about `.agekey`? the Age keygen doesn't have opinions about the file extension, but SOPS does... as `txt` is hoardcoded.

Reply:
Think `.agekey` is an OK suggestion, SOPS guards itself against collisions by looking for a `keys.txt` file in a specific (age) folder.